### DISCUSSION TITLE
How to connect Argo CD to Bitbucket Cloud private workspace without creating a fake user?

### DISCUSSION THREAD

**Discussion by Spenhouet** [argoproj/argo-cd#7671] [Q&A]
We have a company workspace in Bitbucket Cloud with many Git Repositories.
We want to give Argo access to these repositories without needing to create a fake user (which we would need to pay for).

In Codefresh we did set this up with OAuth2 but we do not see that option with Argo.
The Argo docs only explain the connection via HTTPS credentials, tokens or SSH: https://argo-cd.readthedocs.io/en/stable/user-guide/private-repositories/
Bitbucket provides OAuth consumers: https://support.atlassian.com/bitbucket-cloud/docs/use-oauth-on-bitbucket-cloud/
Is it possible to make use of that to connect Argo to all repositories?

Or is there no way to do this without an explicit user in Bitbucket Cloud?

**Comment by aki-ks** [comment]
We are running Bitbucket DataCenter. There's the option to grant repository-accounts for ssh-keys in the settings of each Project.

![Unbenannt](https://user-images.githubusercontent.com/36778587/145111507-63215a2f-028f-4fd7-9f9c-8df929a90a38.png)
t.
